backend/src/common/middlewares/admin-route.middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { type preHandlerHookHandler } from 'fastify';
+
+import { ErrorMessages, UserRole } from '../enums/enums.js';
+import { HttpCode, HttpError } from '../http/http.js';
+
+const adminRoute: preHandlerHookHandler = (request, reply, done) => {
+    const { user } = request;
+
+    const isAdmin = user?.role === UserRole.ADMIN;
+
+    if (!isAdmin) {
+        throw new HttpError({
+            message: ErrorMessages.ROUTE_ADMIN_ONLY,
+            status: HttpCode.FORBIDDEN,
+        });
+    }
+
+    done();
+};
+
```

Comment:
You don't need this in this PR. This is another task. Please don't go beyond the scope of your task !